### PR TITLE
Fixed `remove` method to accept `nullptr` node.

### PR DESCRIPTION
**Pull request by serges147**
Also:
- Clang-Tidy & Sonar fixes (identified during integration into libcyphal project).
- Increased coverage:
image

### CODE REVIEW

File: c++/cavl.hpp [hunk 1]
```diff
@@ -32,13 +34,17 @@
...
 #ifndef CAVL_ASSERT
 #    if defined(CAVL_NO_ASSERT) && CAVL_NO_ASSERT
-#        define CAVL_ASSERT(x) (void) 0
+#                                       // NOLINTNEXTLINE(cppcoreguidelines-macro-usage) function-like macro
+#        define CAVL_ASSERT(x) (void) 0 /* NOSONAR cpp:S960 */
 #    else
-#        include <cassert>  // NOLINTNEXTLINE function-like macro
-#        define CAVL_ASSERT(x) assert(x)
+#        include <cassert>
+#                                        // NOLINTNEXTLINE(cppcoreguidelines-macro-usage) function-like macro
```
Comment (pavel-kirienko) [inline c++/cavl.hpp:41]: Please remove the leading `#` here and on line 37

image

**Review by pavel-kirienko — APPROVED** [review]

File: c++/test.cpp [hunk 2]
```diff
@@ -1,22 +1,23 @@
...
+#include <limits>
 #include <memory>
 #include <numeric>
-#include <limits>
-#include <vector>
+#include <sstream>
+#include <string>
 #include <type_traits>
-#include <functional>
+#include <unity.h>
```
[the remark this answers is not in the document]
Comment (serges147) [reply c++/test.cpp]: @pavel-kirienko Please, two questions on that:
1. why it should go before std includes?
2. should it be even before `#include "cavl.hpp"` (inside of it we also include std headers)?
Thanks!

[the remark this answers is not in the document]
Comment (pavel-kirienko) [reply c++/test.cpp]: The conventional inclusion order is:

- The header associated with the current translation unit.
- Other project headers.
- Third-party headers.
- Standard library includes.

[the remark this answers is not in the document]
Comment (serges147) [reply c++/test.cpp]: ok, I just thought that there was something (magic) about `<unitity>` specifically... ;)

**Pull request merged**